pennylane/qchem/tapering.py

```diff
@@ -598,7 +598,7 @@ def _build_generator(operation, wire_order, op_gen=None):
                     f"Generator for {operation} is not implemented, please provide it with 'op_gen' args."
                 ) from exc
     else:  # check that user-provided generator is correct
-        if not isinstance(op_gen, (qml.Hamiltonian, PauliSentence)) and not isinstance(
+        if not isinstance(op_gen, (qml.ops.Hamiltonian, PauliSentence)) and not isinstance(
```

Comment:
I spent such a long time looking at this because of the double negatives, what if we rearrange it? 😅 But also not the point of this PR. 

More to the point: why would we treat `Hamiltonian` differently than `LinearCombination` here? What if it's a `LinearCombination` with no `pauli_rep`? As-is, a legacy `Hamiltonian` with no `pauli_rep` will be okay, but the same operator expressed as a `LinearCombination` will not. These, for example:

```
op_gen = qml.ops.LinearCombination([2, 3], [qml.X(0), qml.Hadamard(1)])
op_gen_legacy = qml.ops.Hamiltonian([2, 3], [qml.X(0), qml.Hadamard(1)])
```

```suggestion
        if not isinstance(op_gen, (qml.ops.Hamiltonian, qml.ops.LinearCombination, PauliSentence)) and not isinstance(
```

I don't know anything about the `tapering` module, so not sure how significant this is, or whether we even want an operator like the one above to pass. But it does seem to behave differently where I wouldn't expect it to.

Reply:
I don't know much about the `tapering` module either, but my understanding is that this check requires the `pauli_rep` to be a `PauliSentence`, so if something does have a `pauli_rep` it should be checked here. Not sure about why `Hamiltonian` gets past this check though. I don't want to change any existing logic.

Reply:
Ok yeah I understand what you mean now.

Reply:
I've previously thought that a `LinearCombination` would always have a `pauli_rep` and thus included in the second half of the if statement.